lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -570,4 +573,108 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    @Override
+    public StatusEvent updateEvent(
+            StatusEvent ev)
+            throws IndexingException,
+                   RetriableIndexingException {
+
+        // latest states to update
+        StatusEventType latestEventType = ev.getEventType();
+        Boolean latestIsPublic = null;
+        String latestName = null;
+
+        if( ev.isPublic().isPresent()) {
+            latestIsPublic = ev.isPublic().get().booleanValue();
+        }
+        if( ev.getNewName().isPresent()) {
+            latestName = ev.getNewName().get().toString();
+        }
+
+
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        long wsid = Long.valueOf(ev.getAccessGroupId().get().toString()).longValue();
```

Comment:
So officially not all events have access group ids seeing as it's optional. All workspace generated events do have AGIDs. I'm wondering if the optional AGIDs should change...

If it doesn't this code needs to handle that though.

Reply:
The reason for the optional AGIDs was that we may add data sources to the index in the future that don't have access groups, IIRC.